core/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -750,6 +769,58 @@ class DeltaAnalysis(session: SparkSession)
       !SchemaUtils.isReadCompatible(schema.asNullable, existingSchemaOutput.toStructType)
   }
 
+  private def insertIntoByNameMissingColumn(
+    query: LogicalPlan,
+    targetAttrs: Seq[Attribute],
+    deltaTable: DeltaTableV2): Unit = {
+    if (query.output.length < targetAttrs.length) {
+      // Some columns are not specified. We don't allow schema evolution in INSERT INTO BY NAME, so
+      // we need to ensure the missing columns must be generated columns.
+      val userSpecifiedNames = if (session.sessionState.conf.caseSensitiveAnalysis) {
+        query.output.map(a => (a.name, a)).toMap
+      } else {
+        CaseInsensitiveMap(query.output.map(a => (a.name, a)).toMap)
+      }
+      val tableSchema = deltaTable.snapshot.metadata.schema
+      if (tableSchema.length != targetAttrs.length) {
+        // The target attributes may contain the metadata columns by design. Throwing an exception
+        // here in case target attributes may have the metadata columns for Delta in future.
+        throw DeltaErrors.schemaNotConsistentWithTarget(s"$tableSchema", s"$targetAttrs")
+      }
+      deltaTable.snapshot.metadata.schema.foreach { col =>
+        if (!userSpecifiedNames.contains(col.name) &&
+          !ColumnWithDefaultExprUtils.columnHasDefaultExpr(deltaTable.snapshot.protocol, col)) {
+          throw DeltaErrors.missingColumnsInInsertInto(col.name)
+        }
+      }
+    }
+  }
+
+  /**
+   * With Delta, we ACCEPT_ANY_SCHEMA, meaning that Spark doesn't automatically adjust the schema
+   * of INSERT INTO. Here we check if we need to perform any schema adjustment for INSERT INTO by
+   * name queries. We also check that any columns not in the list of user-specified columns must
+   * have a default expression.
+   */
+  private def needsSchemaAdjustmentByName(query: LogicalPlan, targetAttrs: Seq[Attribute],
+      deltaTable: DeltaTableV2): Boolean = {
+    // TODO: update this to allow columns with default expressions to not be
```

Comment:
This will come in a follow-up PR with tests

Reply:
What does this mean? Are there any current tests covering this function... or is the entire thing TODO?

Reply:
The last thing we want is some half-incorrectly-implemented functionality.

Reply:
We add the code to support the new functionality (not specifying generated columns in insert into by name statements), but we don't allow it in this PR. Basically in Spark 3.3, insert into by name only worked when all the columns were specified and otherwise would throw an error. We maintain that behavior here (lines 809-812) so no change in behavior (so existing tests should suffice.)

Once we remove the check on 809-812 we will have the full support for insert into with generated columns. I'll remove it and add the tests in a follow-up PR.

Reply:
Since we have to now do schema validation / evolution for by-name queries it's easier to add all this code now, but artificially block new functionality so the tests can come in a follow-up PR and lessen the amount of code in this PR.